pkg/apps/strategy/support/hooks_test.go

```diff
@@ -0,0 +1,125 @@
+package support
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openshift/origin/pkg/api/apihelpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/fake"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	kapi "k8s.io/kubernetes/pkg/apis/core"
+
+	_ "github.com/openshift/origin/pkg/api/install"
+	appsapi "github.com/openshift/origin/pkg/apps/apis/apps"
+	appstest "github.com/openshift/origin/pkg/apps/apis/apps/test"
+	appsutil "github.com/openshift/origin/pkg/apps/util"
+)
+
+func init() {
+	// Override the streaming handler for the lifecyclePodHook
+	streamFn = func(client typedcorev1.PodInterface, hookPodName string) (io.ReadCloser, error) {
+		return ioutil.NopCloser(strings.NewReader("sample log\n")), nil
+	}
+}
+
+func TestRunHook(t *testing.T) {
```

Comment:
TODO: add more tests, currently this covers ~60% of the lifecycle executor code. This also make sure there are no races.